pylint/checkers/stdlib.py

```diff
@@ -466,27 +466,36 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
 
     def __init__(self, linter: Optional["PyLinter"] = None) -> None:
         BaseChecker.__init__(self, linter)
+        self._deprecated_names_populated = False
         self._deprecated_methods: Set[Any] = set()
+        self._deprecated_attributes: Dict = {}
+        self._deprecated_classes: Dict = {}
+        self._deprecated_modules: Set[Any] = set()
+        self._deprecated_decorators: Set[Any] = set()
+
+    def _ensure_deprecated_names_populated(self):
+        if self._deprecated_names_populated:
+            return
+        # In all versions
         self._deprecated_methods.update(DEPRECATED_METHODS[0])
-        for since_vers, func_list in DEPRECATED_METHODS[sys.version_info[0]].items():
-            if since_vers <= sys.version_info:
+        # Pad (X.Y) -> (X.Y.0), since the constants are 3-tuples
+        py_version = (*get_global_option(self, "py-version"), 0)
```

Comment:
I don't think padding is necessary. Tuple comparison works even with tuples of different length.
```py
>>> (3, 8, 0) >= (3, 8)
True
```

Reply:
This is what I was running into:
```python
>>> since_vers = (3, 9, 0)
>>> py_version = (3, 9)
>>> since_vers <= py_version
False
>>> py_version_padded = (3, 9, 0)
>>> since_vers <= py_version_padded
True
```

Reply:
Ah, but maybe I could rewrite the inequalities instead?

Reply:
I see. In that case it's probably easiest to leave it as is, with padding. Just keep in mind that `py-version = 3.6.2` would  also be valid. AFAICT padding would work here too, though.